### ISSUE TITLE
Envoy proxy for CentOS

### ISSUE THREAD

**Issue by krishamb** [envoyproxy/envoy#5742]
**Issue Template**

*Title*: Envoy Proxy support for CentOS*

*Description*:
  I am trying to compile envoy proxy code using the scripts. It pulls the Centos image from dockerhub but after that I get pcap group not found and no build scripts related to Centos. Ubuntu 16.04 has all the build scripts needed. Does it mean support is dropped for CentOS ? Greatly appreciate your help.

[optional *Relevant Links*:]
  
*Bug Template*

*Title*: Support for CentOS linux distro*

*Description*:
     Envoy proxy should support linux CentOS distro.

*Repro steps*:
  Cent OS

>**Note**: The [Envoy_collect tool](https://github.com/envoyproxy/envoy/blob/master/tools/envoy_collect/README.md)
gathers a tarball with debug logs, config and the following admin
endpoints: /stats, /clusters and /server_info. Please note if there are
privacy concerns, sanitize the data prior to sharing the tarball/pasting.

*Admin and Stats Output*:
>Include the admin output for the following endpoints: /stats,
/clusters, /routes, /server_info. For more information, refer to the
[admin endpoint documentation.](https://www.envoyproxy.io/docs/envoy/latest/operations/admin)

>**Note**: If there are privacy concerns, sanitize the data prior to
sharing.

*Config*:
>Include the config used to configure Envoy.

*Logs*:
>Include the access logs and the Envoy logs.

>**Note**: If there are privacy concerns, sanitize the data prior to
sharing.

*Call Stack*:
> If the Envoy binary is crashing, a call stack is **required**.
Please refer to the [Bazel Stack trace documentation](https://github.com/envoyproxy/envoy/tree/master/bazel#stack-trace-symbol-resolution).

**Comment by mattklein123** [comment]
The CentOS build containers were deleted because no one was maintaining them. We are looking for someone (or multiple people) who are willing to do this.

**Comment by krishamb** [comment]
thank you very much matt. I am volunteering myself for CentOS.

**Comment by snowp** [comment]
There was an attempt at bringing it back in #5262 by @uromahn that might be a useful starting point.

**Comment by krishamb** [comment]
Thank you very much snow. I dont find the build scripts related to centos in master git..I am also getting pcap group not found. I dont create pcap group before adding my user id. Ubuntu has all the scripts for build and environment.

Ambarish

Get Outlook for Android<https://aka.ms/ghei36>

**Comment by lizan** [comment]
I do have an in-house CentOS build and happy to upstream if someone want to co-maintain :)

We can start with an image can only build release (i.e. without asan/tsan etc), I guess that is most are interested?

**Comment by krishamb** [comment]
Does the image have pcap group and build scripts for centOS ?

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by snowp** [comment]
@lizan We'd benefit from a CentOS 7 build image, so I'd be up to help maintain it. Our immediate need is limited to using it to build a release + test our extensions.

**Comment by lizan** [comment]
@snowp Great, then let's pick up this. I will try pull out our image build script into a PR in a few days.

**Comment by krishamb** [comment]
That will help. I can also help out, if needed. Thanks.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by Mydaymatt** [comment]
I am running tomcat and MariaDB on docker container, can i add envoy on top of tomcat? Is there any example?

Thank you in advance :)

**Comment by cube2222** [comment]
Hey, has this moved forward somehow?

**Comment by uromahn** [comment]
Since this has been dragging on for a while and no obvious progress, I think it is time to comment on it for some clarification.

1. Technically, we resurrected a way to build Envoy for CentOS.
1. One issue left to solve is how to best merge it back into the upstream code and keep maintaining it

Some background:
The build toolchain on Ubuntu, which is the primary development environment, is much more modern than what is available on CentOS "out of the box". I was able to upgrade most of it to versions that would support a production build on CentOS. However, not everything is working on CentOS, especially two unit tests are failing due to some differences between Ubuntu and CentOS caused by the much older kernel in CentOS.

I am trying to collaborate with @lizan to get this wrapped up but since both of us have full-time jobs at different employers, @lizan's company requires me to sign an NDA with his employer which my legal department wants to check. Obviously and unfortunately, those "legal matters" quite often take a significant amount of time.

So, I have to ask everyone waiting for a CentOS build to be a bit more patient.
